generators/chipyard/src/main/scala/example/TutorialTile.scala

```diff
@@ -0,0 +1,199 @@
+  val haveBasicCounters: Boolean = true
+  val haveFSDirty: Boolean = false
+  val misaWritable: Boolean = false
+  val haveCFlush: Boolean = false
+  val nL2TLBEntries: Int = 512 // copied from Rocket
+  val mtvecInit: Option[BigInt] = Some(BigInt(0)) // copied from Rocket
+  val mtvecWritable: Boolean = true // copied from Rocket
+  val instBits: Int = if (useCompressed) 16 else 32
+  val lrscCycles: Int = 80 // copied from Rocket
+  val decodeWidth: Int = 1 // TODO: Check
+  val fetchWidth: Int = 1 // TODO: Check
+  val retireWidth: Int = 2
+}
+
+// DOC include start: CanAttachTile
+case class MyTileAttachParams(
+  tileParams: MyTileParams,
+  crossingParams: RocketCrossingParams
+) extends CanAttachTile {
+  type TileType = MyTile
+  val lookup = PriorityMuxHartIdFromSeq(Seq(tileParams))
+}
+// DOC include end: CanAttachTile
+
+case class MyTileParams(
+  name: Option[String] = Some("my_tile"),
+  hartId: Int = 0,
+  trace: Boolean = false,
+  val core: MyCoreParams = MyCoreParams()
+) extends InstantiableTileParams[MyTile]
+{
+  val beuAddr: Option[BigInt] = None
+  val blockerCtrlAddr: Option[BigInt] = None
+  val btb: Option[BTBParams] = Some(BTBParams())
+  val boundaryBuffers: Boolean = false
+  val dcache: Option[DCacheParams] = Some(DCacheParams())
+  val icache: Option[ICacheParams] = Some(ICacheParams())
+  def instantiate(crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)(implicit p: Parameters): MyTile = {
+    new MyTile(this, crossing, lookup)
+  }
+}
+
+// DOC include start: Tile class
+class MyTile(
+  val myParams: MyTileParams,
+  crossing: ClockCrossingType,
+  lookup: LookupByHartIdImpl,
+  q: Parameters)
+  extends BaseTile(myParams, crossing, lookup, q)
+  with SinksExternalInterrupts
+  with SourcesExternalNotifications
+{
+
+  // Private constructor ensures altered LazyModule.p is used implicitly
+  def this(params: MyTileParams, crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)(implicit p: Parameters) =
+    this(params, crossing.crossingType, lookup, p)
+
+  // Require TileLink nodes
+  val intOutwardNode = IntIdentityNode()
+  val masterNode = visibilityNode
+  val slaveNode = TLIdentityNode()
+
+  // Implementation class (See below)
+  override lazy val module = new MyTileModuleImp(this)
+
+  // Required entry of CPU device in the device tree for interrupt purpose
+  val cpuDevice: SimpleDevice = new SimpleDevice("cpu", Seq("my-organization,my-cpu", "riscv")) {
+    override def parent = Some(ResourceAnchors.cpus)
+    override def describe(resources: ResourceBindings): Description = {
+      val Description(name, mapping) = super.describe(resources)
+      Description(name, mapping ++
+                        cpuProperties ++
+                        nextLevelCacheProperty ++
+                        tileProperties)
+    }
+  }
+
+  ResourceBinding {
+    Resource(cpuDevice, "reg").bind(ResourceAddress(hartId))
+  }
+
+  // (Connection to bus, interrupt, etc.)
+// }
```

Comment:
?

Reply:
I couldn't skip the section of the tile class body which I intended to put it in another section. to explain how to create TileLink nodes and connect them to the bus, which I think will distract the users if I put it here. Not sure how to add that right bracket in RST so that they are in the same code section since ``literalinclude`` cannot have extra code following it.

Reply:
Just don't include the commented out bracket. Its fine if the included code in the documentation is missing an end bracket

Reply:
Please remove these commented out brackets. Its ok if the inlined code section contains only part of a class.

Reply:
Fixed. I forgot to look at the old comments last time.